src/main/kotlin/kweb/html/ElementReader.kt

```diff
@@ -11,7 +12,9 @@ open class ElementReader(protected val receiver: WebBrowser, internal val elemen
     constructor(element: Element) : this(element.browser, element.id)
 
     init {
-        require(!receiver.kweb.isCatchingOutbound()) {
+        //TODO I'm not sure if we want to allow reading the DOM during a render or non immediate event
+        //require(receiver.kweb.isCatchingOutbound() != Kweb.CatcherType.IMMEDIATE_EVENT)
+        require(receiver.kweb.isCatchingOutbound() == null) {
             """
             Reading the DOM when an outboundMessageCatcher is set is likely to have unintended consequences.
             Most likely you are trying to read the DOM within an `immediatelyOn {...}` block.
```

Comment:
Nitpick but `immediatelyOn` is an old syntax, should be `onImmediate`.

Reply:
I thought this was a little weird. I've read this comment a few times recently, but hadn't seen `immediatelyOn` anywhere else. I'll update it.